packages/gatsby-link/scripts/build.js

```diff
@@ -0,0 +1,31 @@
+const { bundleClientModule } = require(`../../../scripts/bundle-client-module`)
+
+let watch = false
+
+for (const arg of process.argv.slice(2)) {
+  const [key, value] = arg.split(`=`)
+  switch (key) {
+    case `--watch`:
+      watch = value
+      break
+  }
+}
```

Comment:
Leave this logic here in case we need to add more flags in the future (same for each bundle script)